llama_stack/providers/utils/inference/prompt_adapter.py

```diff
@@ -117,27 +116,29 @@ async def _localize_single(c: str | InterleavedContentItem) -> str | RawContentI
         elif isinstance(c, TextContentItem):
             return RawTextItem(text=c.text)
         elif isinstance(c, ImageContentItem):
-            # load image and return PIL version
-            img = c.data
-            if isinstance(img, URL):
-                if img.uri.startswith("data"):
-                    match = re.match(r"data:image/(\w+);base64,(.+)", img.uri)
+            if c.url:
+                # Load image bytes from URL
+                if c.url.uri.startswith("data"):
+                    match = re.match(r"data:image/(\w+);base64,(.+)", c.url.uri)
                     if not match:
                         raise ValueError("Invalid data URL format")
                     _, image_data = match.groups()
                     data = base64.b64decode(image_data)
-                elif img.uri.startswith("file://"):
-                    path = img.uri[len("file://") :]
+                elif c.url.uri.startswith("file://"):
+                    path = c.url.uri[len("file://") :]
                     with open(path, "rb") as f:
                         data = f.read()  # type: ignore
```

Comment:
you would need a try block here?

Reply:
@raghotham why is a `try` block needed here?

Reply:
Maybe it will help to see what error is send back to the client when the read fails. Main reason to add try block is to add some context that the client can consume instead of a raw system error.

Reply:
I guess we could add try blocks in the future to append more context with loggings. Leaving it as pass through for now as we can get details from stack trace.